### ISSUE TITLE
doc: `same_functions_in_if_condition` example contradicts actual lint behavior

### ISSUE THREAD

**Issue by andyrift** [rust-lang/rust-clippy#17259]
The `same_functions_in_if_condition` lint is supposed to check that function is called several times in if conditions

But the "correct" example suggest the exact code that should trigger the lint:
https://github.com/rust-lang/rust-clippy/blob/d7246b03ea4761416cfba19d34708e4200fce604/clippy_lints/src/ifs/mod.rs#L120-L127

The whole doc for this lint looks like a copy-paste error to me.
I was just browsing clippy lints and this one confused me.

Not sure what the correct documentation should look like, therefore an issue.

**Comment by Hamdan-Khan** [comment]
I think this lint is supposed to check for same functions in if condition **with the same comparison operations**. Lint title is kinda misleading.

> But the "correct" example suggest the exact code that should trigger the lint

It shouldn't trigger the lint. You probably overlooked `baz` for `bar` in the other branch. 

```rust
    if foo() == bar {
        ...
    } else if foo() == baz {
        ...
    }
```

**Comment by andyrift** [comment]
@Hamdan-Khan I did see different variables. But the "same condition" is already covered by `ifs_same_cond` lint
